ui/ui-frontend/projects/archive-search/src/app/archive/validators/management-rules-validator.service.ts

```diff
@@ -76,9 +75,13 @@ export class ManagementRulesValidatorService {
       this.managementRules = data;
     });
 
-    if (this.managementRules.findIndex((managementRule) => managementRule.category === RuleTypeEnum.APPRAISALRULE) !== -1) {
+    this.managementRulesSharedDataService.getRuleCategory().subscribe((data) => {
+      this.ruleCategorySelected = data;
+    });
+
+    if (this.managementRules.findIndex((managementRule) => managementRule.category === this.ruleCategorySelected) !== -1) {
       this.ruleActions = this.managementRules.find(
-        (managementRule) => managementRule.category === RuleTypeEnum.APPRAISALRULE
+        (managementRule) => managementRule.category === this.ruleCategorySelected
```

Comment:
C'est pas plus simple de faire un `const managementRule = this.managementRules.find` avant le if et de remplacer le if par un `if (managementRule)` ?

Ca évite de faire deux parcours du tableau de rules.